src/common/proc.c

```diff
@@ -130,45 +129,41 @@ crt_proc_prop_entries(crt_proc_t proc, crt_proc_op_t opc, daos_prop_t *prop)
 }
 
 int
-crt_proc_daos_prop_t(crt_proc_t proc, daos_prop_t **data)
+crt_proc_daos_prop_t(crt_proc_t proc, crt_proc_op_t proc_op, daos_prop_t **data)
 {
 	daos_prop_t		*prop;
-	crt_proc_op_t		 proc_op;
 	uint32_t		 nr = 0, tmp = 0;
 	int			 rc;
 
 	if (proc == NULL || data == NULL)
 		return -DER_INVAL;
 
-	rc = crt_proc_get_op(proc, &proc_op);
-	if (rc)
-		return rc;
 	switch (proc_op) {
 	case CRT_PROC_ENCODE:
 		prop = *data;
 		if (prop == NULL || prop->dpp_nr == 0 ||
 		    prop->dpp_entries == NULL) {
 			nr = 0;
-			rc = crt_proc_uint32_t(proc, &nr);
+			rc = crt_proc_uint32_t(proc, proc_op, &nr);
```

Comment:
Or maybe even crt_proc_uint32_t_enconde(proc, &nr) to avoid another case statement in crt_proc_uint32_t.  Perhaps with enough preprocessor magic you'd be able to inline these and have the compiler optimise proc_op away as it's value is known at this point.

Out of scope for this PR I suspect but there's more to be gained here.